src/components/Todo/Todo.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { useState } from 'react';
+import cn from 'classnames';
+
+import './Todo.scss';
+import { TodoType } from '../../types/TodoType';
+
+type Props = {
+  todo: TodoType;
+  onDelete: (todoId: number) => Promise<void>;
+  ids: number[];
+  updatedStatus: number[];
+  onChangeStatus: (todo: TodoType) => Promise<void>;
+  onChangeTitle: (todo: TodoType) => Promise<void>;
+};
+
+export const Todo: React.FC<Props> = ({
+  todo,
+  onDelete,
+  ids,
+  updatedStatus,
+  onChangeStatus,
+  onChangeTitle,
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [isInput, setIsInput] = useState(false);
+  const [title, setTitle] = useState('');
+
+  function handleLoading<T>(
+    cause: (param: T) => Promise<void>,
+    param: T,
+  ): void {
+    setIsLoading(true);
+
+    cause(param)
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }
+
+  const deleteTodo = () => {
+    handleLoading<number>(onDelete, todo.id);
+  };
+
+  const updateStatus = () => {
+    handleLoading<TodoType>(onChangeStatus, todo);
+  };
+
+  const handleIsInput = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setIsInput(!isInput);
+
+    if (title === '') {
+      setIsLoading(true);
+
+      onDelete(todo.id)
+        .finally(() => {
+          setIsLoading(false);
+        });
+
+      return;
+    }
+
+    if (todo.title !== title) {
+      setIsLoading(true);
+
+      onChangeTitle({ ...todo, title })
+        .finally(() => {
+          setIsLoading(false);
+        });
+    }
+  };
+
+  const onDoubleClick = () => {
+    setIsInput(!isInput);
+    setTitle(todo.title);
+  };
+
+  const handleSetTitle = (event: React.ChangeEvent<HTMLInputElement>) => (
+    setTitle(event.target.value)
+  );
+
+  const resetChanges = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.code === 'Escape') {
+      setIsInput(!isInput);
+      setTitle(todo.title);
+    }
+  };
+
+  return (
+    <div className={cn('todo', { completed: todo.completed })}>
+      <label className="todo__status-label">
+        <input
+          type="checkbox"
+          className="todo__status"
+          onClick={updateStatus}
+        />
+      </label>
+
+      {isInput && (
+        <form
+          onSubmit={handleIsInput}
+          onBlur={handleIsInput}
+        >
+          <input
+            type="text"
+            className="todo__title-field"
+            placeholder="Empty todo will be deleted"
+            value={title}
+            onChange={handleSetTitle}
+            onKeyUp={resetChanges}
+            // eslint-disable-next-line
```

Comment:
Try to fix eslint error and remove comment

Reply:
still not fixed, try to rewrite this logic using `ref` and `useRef`
<img width="723" alt="image" src="https://github.com/mate-academy/react_todo-app-with-api/assets/70220667/9c4ce1f0-36d3-46ae-bbcd-af8fc5fadead">